src/commands/producers.ts

```diff
@@ -67,35 +68,69 @@ export class ProduceRecordCommandHandler {
+        command.clusterId = client.cluster.id;
+        const producerUri = createProducerUri(command);
+        const record = {
+            topic: topicId,
+            messages: messages,
+        };
+        // Start the producer
+        await startProducerWithProgress(producerUri, record, this.producerCollection, channel, times, this.explorer);
+    }
+}
 
-            if (error.message) {
-                channel.appendLine(`Error: ${error.message}`);
-            } else {
-                channel.appendLine(`Error: ${error}`);
-            }
-        }
+async function startProducerWithProgress(producerUri: vscode.Uri, record: ProducerRecord, producerCollection: ProducerCollection, channel: vscode.OutputChannel, times: number, explorer?: KafkaExplorer) {
+    const producer = producerCollection.get(producerUri);
+    if (producer && producer.state === ProducerLaunchState.connecting) {
+        vscode.window.showErrorMessage(`The producer cannot be started because it is producing.`);
+        return;
     }
+    await vscode.window.withProgress({
+        location: vscode.ProgressLocation.Window,
+        title: `Starting producer '${producerUri}'.`,
+        cancellable: false
+    }, (progress, token) => {
+        return new Promise((resolve, reject) => {
+            // 1. Connect the producer
+            producerCollection
+                .create(producerUri)
+                .then(producer => {
+
+                    channel.appendLine(`Producing record(s)`);
+                    const startOperation = performance.now();
+
+                    // 2. Send the producer record.
+                    producerCollection
+                        .send(producerUri, record)
+                        .then(() => {
+
+                            const finishedOperation = performance.now();
+                            const elapsed = (finishedOperation - startOperation).toFixed(2);
+
+                            channel.appendLine(`Produced ${times} record(s) (${elapsed}ms)`);
+
+                            if (explorer) {
+                                explorer.refresh();
+                            }
+
+                            // 3. Disconnect the producer
+                            producerCollection
+                                .disconnect(producerUri)
+                                .then(() => resolve(producer));
+                        })
+                        .catch(error => {
+                            const finishedOperation = performance.now();
+                            const elapsed = (finishedOperation - startOperation).toFixed(2);
+                            channel.appendLine(`Failed to produce record(s) (${elapsed}ms)`);
+                            if (error.message) {
+                                channel.appendLine(`Error: ${error.message}`);
+                            } else {
+                                channel.appendLine(`Error: ${error}`);
+                            }
```

Comment:
should prolly disconnect the producer in case of an error

Reply:
Disconnect producer is a bad idea because producer connect takes some times. In otherwise you connect one time to the producer and you disconnect never.